src/scheduler/gc_work.rs

```diff
@@ -476,6 +476,56 @@ impl<E: ProcessEdgesWork> GCWork<E::VM> for E {
     }
 }
 
+/// A general process edges implementation using SFT. A plan can always implement their own process edges. However,
+/// Most plans can use this work packet for tracing amd they do not need to provide a plan-specific trace object work packet.
+/// If they choose to use this type, they need to provide a correct implementation for some related methods
+/// (such as `Space.set_copy_for_sft_trace()`, `SFT.sft_trace_object()`).
+/// Some plans are not using this type, mostly due to more complex tracing. Either it is impossible to use this type, or
+/// there is performance overheads for using this general trace type. In such cases, they implement their specific process edges.
+pub struct SFTProcessEdges<VM: VMBinding> {
+    pub base: ProcessEdgesBase<VM>,
+}
+
+impl<VM: VMBinding> ProcessEdgesWork for SFTProcessEdges<VM> {
+    type VM = VM;
+    fn new(edges: Vec<Address>, roots: bool, mmtk: &'static MMTK<VM>) -> Self {
+        let base = ProcessEdgesBase::new(edges, roots, mmtk);
+        Self { base }
+    }
+
+    #[inline]
+    fn trace_object(&mut self, object: ObjectReference) -> ObjectReference {
+        use crate::policy::space::*;
+
+        if object.is_null() {
+            return object;
+        }
+
+        // Erase <VM> type parameter
```

Comment:
Is it possible to add a `<VM>` type parameter to `SFTMap`, i.e. `SFTMap<VM>`?  Will that make it unnecessary to do the erasure here?

Maybe the current way of initialising `SFT_MAP` does not allow us to do this.  I am thinking about moving it into `MMTK<VM>` as a field.

Reply:
This is a good point. There are multiple issues here:
1. Should we have `SFTMap<VM>`? 
I think we should if we can. We do not expect different VMs using MMTk in the same process (at least at this point, that is not our goal).
2. Should `SFTMap<VM>` be put to an `MMTK<VM>` instance?
I am not sure, but possibly yes, because it includes pointers to spaces that are in one instance. However, to support multiple `MMTK` instances, we would need to know what address range is used for one instance, and what address range is still available for using. So we will need to have some kind of heap data structure that is shared across instances. Currently, these heap data structures are VM map, Mmapper, and SFT map, and they are intentionally static. But it may be possible that keep some static heap data structure, but move `SFTMap` to `MMTK` -- but I don't know how feasible that is.

Anyway, I don't think we can do this change in this PR. If you think this current hack is not acceptable, I can work on refactoring `SFTMap` first, and then let's revisit this PR.

Reply:
OK. Let's not make the change in this PR, and think about it later.